execution_engine_testing/tests/bin/disk_use.rs

```diff
@@ -0,0 +1,254 @@
+    transfer::create_initial_accounts_and_run_genesis(
+        &mut builder,
+        vec![TARGET_ADDR],
+        purse_amount,
+    );
+
+    let purses = transfer::create_test_purses(
+        &mut builder,
+        *DEFAULT_ACCOUNT_ADDR,
+        purse_count as u64,
+        purse_amount,
+    );
+
+    let exec_requests = transfer::create_multiple_native_transfers_to_purses(
+        *DEFAULT_ACCOUNT_ADDR,
+        transfer_count,
+        &purses,
+    );
+
+    let mut total_transfers = 0;
+    println!("on_disk_size_bytes, transfer_count",);
+
+    total_transfers += exec_requests.len();
+    transfer::transfer_to_account_multiple_native_transfers(&mut builder, &exec_requests, true);
+    println!(
+        "{}, {}",
+        builder.lmdb_on_disk_size().unwrap(),
+        total_transfers
+    );
+
+    let env = builder.lmdb_environment();
+    let db = builder.lmdb_handle();
+    let env = env.env();
+
+    let txn = env.begin_ro_txn()?;
+    let mut cursor = txn.open_ro_cursor(db)?;
+    let mut record_count = 0;
+    let mut largest_record = 0;
+
+    let mut key_tags = HashMap::<String, usize>::new();
+    let mut stored_value_tags = HashMap::<String, usize>::new();
+    let mut trie_value_lengths = HashMap::<(String, String), Vec<usize>>::new();
+    let mut pointer_block_lengths = Vec::new();
+    let mut extension_node_lengths = Vec::new();
+    let mut total_value_bytes_read = 0;
+    let mut total_key_bytes_read = 0;
+
+    for (key, value) in cursor.iter() {
+        // count key len
+        total_key_bytes_read += key.len();
+
+        let (_key, _rest) = Digest::from_bytes(key).expect("should deserialize");
+        let byte_len = value.len();
+
+        // count value len
+        total_value_bytes_read += byte_len;
+
+        let (trie, _remainder) =
+            Trie::<Key, StoredValue>::from_bytes(value).expect("should be a trie");
+
+        match trie {
+            Trie::Leaf { key, value } => {
+                let key_tag = key.type_string();
+                let stored_value_tag = value.type_name();
+
+                *key_tags.entry(key_tag.clone()).or_default() += 1;
+                *stored_value_tags
+                    .entry(stored_value_tag.clone())
+                    .or_default() += 1;
+                let trie_length_values = trie_value_lengths
+                    .entry((key_tag, stored_value_tag))
+                    .or_default();
+                trie_length_values.push(byte_len);
+            }
+            Trie::Node { pointer_block: _ } => {
+                pointer_block_lengths.push(byte_len);
+            }
+            Trie::Extension {
+                affix: _,
+                pointer: _,
+            } => extension_node_lengths.push(byte_len),
+        }
+
+        record_count += 1;
+        let serialized_len = value.len();
+        if largest_record < serialized_len {
+            println!("found new largest DB entry with len {}", serialized_len);
```

Comment:
Won't we print this multiple times - every time we find a new record that is bigger than the previous biggest one? Isn't that a bit too chatty? I think we can drop it and just report at the very end, after the loop, next to other statistics.